src/iotjs.c

```diff
@@ -134,7 +134,27 @@ void iotjs_run(iotjs_environment_t* env) {
                           iotjs_js_modules_l, module_iotjs_idx, 0);
 #endif
 
-  if (jerry_value_is_error(jmain) && !iotjs_environment_is_exiting(env)) {
+  if (jerry_value_is_abort(jmain)) {
+    jerry_value_t abort_value = jerry_get_value_from_error(jmain, false);
```

Comment:
IMHO: This whole code block inside the `if` should be moved into its own method.